File: ios/Classes/BucketeerPluginEvaluationUpdateListener.swift
import Foundation
import Flutter
import Bucketeer

class BucketeerPluginEvaluationUpdateListener: NSObject, FlutterStreamHandler {
    private var eventSink: FlutterEventSink?

    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        self.eventSink = events
        return nil
    }

    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        eventSink = nil
        return nil
    }
}

extension BucketeerPluginEvaluationUpdateListener: EvaluationUpdateListener {
    func onUpdate() {
        if let eventBus = eventSink {
            eventBus(true)
        }
    }
}

File: android/src/main/kotlin/io/bucketeer/sdk/flutter/BucketeerPluginEvaluationUpdateListener.kt
package io.bucketeer.sdk.flutter

import io.bucketeer.sdk.android.BKTClient
import io.flutter.plugin.common.EventChannel

class BucketeerPluginEvaluationUpdateListener : BKTClient.EvaluationUpdateListener,
  EventChannel.StreamHandler {

  private var eventSink: EventChannel.EventSink? = null

  override fun onUpdate() {
    eventSink?.success(true)
  }

  override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
    eventSink = events
  }

  override fun onCancel(arguments: Any?) {
    eventSink = null
  }
}